src/components/PromotedActionsBar.tsx

```diff
@@ -0,0 +1,147 @@
+import Button from './Button';
+import ConfirmModal from './ConfirmModal';
+import type {ThreeDotsMenuItem} from './HeaderWithBackButton/types';
+import * as Expensicons from './Icon/Expensicons';
+
+type PromotedAction = {
+    key: string;
+} & ThreeDotsMenuItem;
+
+type ReportPromotedAction = (report: OnyxReport) => PromotedAction;
+
+type PromotedActionsType = {
+    pin: ReportPromotedAction;
+    // message: (accountID: number) => PromotedAction;
+    // join: ReportPromotedAction;
+    // share: ReportPromotedAction;
+    // hold: () => PromotedAction;
+};
+
+const PromotedActions = {
+    pin: (report) => ({
+        key: 'pin',
+        ...HeaderUtils.getPinMenuItem(report),
+    }),
+    // TODO: Uncomment the following lines when the corresponding features are implemented
+    // message: (accountID) => ({
+    //     key: 'message',
+    //     icon: Expensicons.CommentBubbles,
+    //     text: Localize.translateLocal('common.message'),
+    //     onSelected: () => ReportActions.navigateToAndOpenReportWithAccountIDs([accountID]),
+    // }),
+    // join: (report) => ({
+    //     key: 'join',
+    //     icon: Expensicons.CommentBubbles,
+    //     text: Localize.translateLocal('common.join'),
+    //     onSelected: () => Session.checkIfActionIsAllowed(() => Report.joinRoom(report)),
+    // }),
+    // share: (report) => ({
+    //     key: 'share',
+    //     icon: Expensicons.QrCode,
+    //     text: Localize.translateLocal('common.share'),
+    //     onSelected: () => Navigation.navigate(ROUTES.REPORT_WITH_ID_DETAILS_SHARE_CODE.getRoute(report?.reportID ?? '')),
+    // }),
+    // hold: () => ({
+    //     key: 'hold',
+    //     icon: Expensicons.Stopwatch,
+    //     text: Localize.translateLocal('iou.hold'),
+    //     onSelected: () => {},
+    // }),
+} satisfies PromotedActionsType;
+
+type PromotedActionsBarProps = {
+    /** The report of actions */
+    report?: OnyxReport;
+
+    /** The list of actions to show */
+    promotedActions: PromotedAction[];
+
+    /** The style of the container */
+    containerStyle?: StyleProp<ViewStyle>;
+
+    /**
+     * Whether to show the `Leave` button.
+     * @deprecated Remove this prop when @src/pages/ReportDetailsPage.tsx is updated
+     */
+    shouldShowLeaveButton?: boolean;
+};
+
+function PromotedActionsBar({report, promotedActions, containerStyle, shouldShowLeaveButton}: PromotedActionsBarProps) {
+    const theme = useTheme();
+    const styles = useThemeStyles();
+    const [isLastMemberLeavingGroupModalVisible, setIsLastMemberLeavingGroupModalVisible] = useState(false);
+    const {translate} = useLocalize();
+
+    if (promotedActions.length === 0) {
+        return null;
+    }
+
+    return (
+        <View style={[styles.flexRow, styles.ph5, styles.mb5, styles.gap2, styles.mw100, styles.w100, styles.justifyContentCenter, containerStyle]}>
+            {/* TODO: Remove the `Leave` button when @src/pages/ReportDetailsPage.tsx is updated */}
```

Comment:
Do we have a WIP PR for this yet?

Reply:
I’m handling this one in my ReportDetailsPage PR, I’ll link it here after I put up a draft PR for it later in the evening.

Reply:
Please check [this PR](https://github.com/Expensify/App/pull/41823#discussion_r1618315934) we're already moving it there.